### PR TITLE
Add #country_regions endpoint

### PR DESCRIPTION
**Pull request by RadekMolenda**
This endpoint will allow users to fetch all the administrative regions in one 
request. This sometimes might be quite useful

### CODE REVIEW

**Comment by ChuckJHardy** [comment]
:+1:

**Comment by mfilej** [comment]
I have figured out why we can rename fcode! Give me a few minutes.

**Comment by mfilej** [comment]
Ok so we said we want to keep `fcode` because that is what is returned by the api. But! We have this layer where we get do dictate the results as well: https://github.com/Leadformance/gemonames/blob/master/lib/gemonames/api_client.rb#L135-L143

So if in this layer we make sure that we return an attribute with a name that actually makes sense, we can use the same attribute name as input.

I think this is important in this case because `fcode` means nothing and it is not even documented in their docs. I think using `admin_code` instead would make a lot of sense.

Wdyt?

(If you don't want to deal with it I'm happy to do the necessary changes). :smile:

**Comment by mfilej** [comment]
@RadekMolenda Updated to please CodeClimate. I thought memoizing was overkill for the null objects so I just wrapped them in lambdas. Curious to hear your thoughts.

**Comment by RadekMolenda** [comment]
:+1:

File: lib/gemonames/results.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    :admin_id3,
+    :admin_id2,
+    :admin_id1,
+    :country_id,
+    :result,
+  ) do
+    alias_method :result?, :result
+    alias_method :present?, :result
+
+    def self.with_nil
```
Comment (mfilej) [inline lib/gemonames/results.rb:17]: I wasn't sure about `with_nil` vs `with_nils`. Both sound off to be honest, but I like the symmetry with `with` which is what we use for non-null objects.

Comment (mfilej) [inline lib/gemonames/results.rb:17]: I also considered plain `null`.

**Pull request merged**